app/code/Magento/LoginAsCustomerAssistance/Plugin/LoginAsCustomerButtonDataProviderPlugin.php

```diff
@@ -40,7 +48,8 @@ public function __construct(
      */
     public function afterGetData(DataProvider $subject, array $result, int $customerId): array
     {
-        if (isset($result['on_click']) && !$this->isAssistanceEnabled->execute($customerId)) {
+        if (isset($result['on_click']) &&
+            !($this->isAssistanceEnabled->execute($customerId) && $this->isCustomerEnabled->execute($customerId))) {
             $result['on_click'] = 'window.lacNotAllowedPopup()';
         }
```

Comment:
In my opinion it makes more sense to firstly check if the customer is enabled and then do the rest of checking. 
```suggestion
        if (isset($result['on_click']) 
            && !($this->isCustomerEnabled->execute($customerId) && $this->isAssistanceEnabled->execute($customerId))
        ) {
            $result['on_click'] = 'window.lacNotAllowedPopup()';
        }
```